x-pack/plugins/upgrade_assistant/server/lib/telemetry/usage_collector.ts

```diff
@@ -85,11 +85,8 @@ export async function fetchUpgradeAssistantMetrics(
     }
 
     const telemetryObj = defaultTelemetrySavedObject;
-    Object.keys(defaultTelemetrySavedObject).forEach((key: string) => {
-      if (
-        !upgradeAssistantTelemetrySavedObjectAttrs.hasOwnProperty(key) ||
-        !upgradeAssistantTelemetrySavedObjectAttrs[key]
-      ) {
+    Object.keys(upgradeAssistantTelemetrySavedObjectAttrs).forEach((key: string) => {
+      if (!has(defaultTelemetrySavedObject, key)) {
```

Comment:
I'm afraid this won't work the way you'd expect it to. `_.has` can return `true` for properties that aren't "own properties", even though that's what's being claimed.

I'd really prefer we use an approach which does not take advantage of lodash here, so that we can make sure these objects are merged correctly going forward

Reply:
I think that is fixed now @legrego